app/controllers/users/unlocks_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::UnlocksController < Devise::UnlocksController
```

Comment:
I don't think you need all these empty controllers. Devise has a way to override a single controller instead of all of them.

https://stackoverflow.com/a/3553561/2425788

Reply:
Can i just delete all the others controllers? I added some functions in devise to permit more parameters and I don if it still works